main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -1856,18 +1875,29 @@ object Parser2 {
       close(mark, TreeKind.Expr.Unary)
     }
 
+    /**
+      * The condition and then sub-trees will always exist, but the else sub-tree is optional.
+      * If condition/then branch is missing in the source code, it will be represented as an Expr.Error inside that sub-tree.
+      */
     private def ifThenElseExpr()(implicit s: State): Mark.Closed = {
       assert(at(TokenKind.KeywordIf))
       val mark = open()
       expect(TokenKind.KeywordIf, SyntacticContext.Expr.OtherExpr)
       expect(TokenKind.ParenL, SyntacticContext.Expr.OtherExpr)
-      expression()
+      val exitTokens: Set[TokenKind] = Set(TokenKind.ParenR, TokenKind.KeywordElse)
+      val condMark = open()
+      expression(exitTokens = exitTokens)
```

Comment:
For this call the exitTokens should only be `ParenR` and in the next call it should only be `Then`